### PR TITLE
Feature #72: Create endpoint to return survey analytics

### PR DESCRIPTION
**Pull request by pathumd**
### Summary of changes

- Implemented the /user/survey/<surveyId>/results endpoint, which accepts a GET request
- Removed stub in the frontend and added call to this endpoint

### Structure of the returned JSON

`
{
    "responses": [
        {
            "question": "Fav game",
            "responses": [
                {
                    "Fortnite": 3
                },
                {
                    "Valorant": 1
                },
                {
                    "Apex Legends": 1
                }
            ],
            "questionType": "ChoiceQuestion"
        },
        {
            "question": "Fav number",
            "responses": [
                {
                    "50": 3
                },
                {
                    "100": 1
                },
                {
                    "70": 1
                }
            ],
            "questionType": "NumberQuestion"
        },
        {
            "question": "Name",
            "responses": [
                "Pathum Danthanarayana",
                "Pathum D",
                "Pathum",
                "Pathum",
                "Pathum"
            ],
            "questionType": "TextQuestion"
        }
    ],
    "description": "Test"
}
`

- In short, the returned JSON is structured so that the survey description, and a list of responses is returned to the frontend. Specifically, each item in this list models a question in the survey, which also contains a list of responses. For questions that are of type NumberQuestion and ChoiceQuestion, each element in this list is a HashMap that uses the string representation of the response itself as the key, and the value is the total count of that response in the survey. A HashMap 'entry' is created for each unique (number or choice) response. For TextQuestions, just a list of all the responses will be provided. In other words, analytics are only created for NumberQuestions and ChoiceQuestions.

### Visuals

Screenshot 2024-11-28 at 2 09 15 PM
Screenshot 2024-11-28 at 2 09 24 PM

### CODE REVIEW

**Comment by pathumd** [comment]
Note: This endpoint has not been formally tested, as @MichaelKyrollos will be writing a test for this endpoint

File: backend/src/main/java/sysc4806group25/monkeypoll/controller/SurveyController.java [hunk 1]
```diff
@@ -126,6 +124,131 @@ public ResponseEntity<?> getSurveyQuestionsAndResponses(@PathVariable long surve
...
+                        int count = 0;
+                        // Compute number of occurrences
+                        for (ChoiceResponse innerResponse : allChoiceResponses) {
+                            // Note: Comparing ChoiceOptions by choiceOptionId (ChoiceOption PK)
+                            if (response.getResponse().getChoiceOptionId() == innerResponse.getResponse().getChoiceOptionId()) {
+                                // Match found
+                                count++;
+                            }
+                        }
+                        // Create the HashMap 'entry' and add to list
```
Comment (MichaelKyrollos) [inline backend/src/main/java/sysc4806group25/monkeypoll/controller/SurveyController.java:231]: Good work with the comments

**Review by yehandesilva — APPROVED** [review]
Looks good. Removed stub as required.

**Review by MichaelKyrollos — APPROVED** [review]
Looks good, I tested it out on my local computer and everything seems to be working. Looking forward to seeing the frontend all working!
